url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/quartz/triggers/SimpleTriggerImpl.java

```diff
@@ -0,0 +1,605 @@
+   * @exception IllegalArgumentException if repeatCount is < 0
+   */
+  public void setRepeatCount(int repeatCount) {
+
+    if (repeatCount < 0 && repeatCount != REPEAT_INDEFINITELY) {
+      throw new IllegalArgumentException("Repeat count must be >= 0, use the " + "constant REPEAT_INDEFINITELY for infinite.");
+    }
+
+    this.repeatCount = repeatCount;
+  }
+
+  @Override
+  public long getRepeatInterval() {
+
+    return repeatInterval;
+  }
+
+  /**
+   * <p>
+   * Set the the time interval (in milliseconds) at which the <code>SimpleTrigger</code> should repeat.
+   * </p>
+   *
+   * @exception IllegalArgumentException if repeatInterval is <= 0
+   */
+  public void setRepeatInterval(long repeatInterval) {
+
+    if (repeatInterval < 0) {
+      throw new IllegalArgumentException("Repeat interval must be >= 0");
+    }
+
+    this.repeatInterval = repeatInterval;
+  }
+
+  /**
+   * <p>
+   * Get the number of times the <code>SimpleTrigger</code> has already fired.
+   * </p>
+   */
+  @Override
+  public int getTimesTriggered() {
+
+    return timesTriggered;
+  }
+
+  /**
+   * <p>
+   * Set the number of times the <code>SimpleTrigger</code> has already fired.
+   * </p>
+   */
+  public void setTimesTriggered(int timesTriggered) {
+
+    this.timesTriggered = timesTriggered;
+  }
+
+  @Override
+  protected boolean validateMisfireInstruction(int misfireInstruction) {
+
+    if (misfireInstruction < MISFIRE_INSTRUCTION_SMART_POLICY) {
+      return false;
+    }
+
+    if (misfireInstruction > MISFIRE_INSTRUCTION_RESCHEDULE_NEXT_WITH_EXISTING_COUNT) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * <p>
+   * Updates the <code>SimpleTrigger</code>'s state based on the MISFIRE_INSTRUCTION_XXX that was selected when the <code>SimpleTrigger</code> was
+   * created.
+   * </p>
+   * <p>
+   * If the misfire instruction is set to MISFIRE_INSTRUCTION_SMART_POLICY, then the following scheme will be used: <br>
+   * <ul>
+   * <li>If the Repeat Count is <code>0</code>, then the instruction will be interpreted as <code>MISFIRE_INSTRUCTION_FIRE_NOW</code>.</li>
+   * <li>If the Repeat Count is <code>REPEAT_INDEFINITELY</code>, then the instruction will be interpreted as
+   * <code>MISFIRE_INSTRUCTION_RESCHEDULE_NEXT_WITH_REMAINING_COUNT</code>. <b>WARNING:</b> using
+   * MISFIRE_INSTRUCTION_RESCHEDULE_NEXT_WITH_REMAINING_COUNT with a trigger that has a non-null end-time may cause the trigger to never fire again if
+   * the end-time arrived during the misfire time span.</li>
+   * <li>If the Repeat Count is <code>&gt; 0</code>, then the instruction will be interpreted as
+   * <code>MISFIRE_INSTRUCTION_RESCHEDULE_NOW_WITH_EXISTING_REPEAT_COUNT</code>.</li>
+   * </ul>
+   * </p>
+   * <p>
+   * If the misfire instruction is set to <code>Trigger.MISFIRE_INSTRUCTION_SKIP_TO_NEXT_FIRE_AFTER_CURRENT_DATE</code> then the behavior will be
+   * identical to
+   */
+  @Override
+  public void updateAfterMisfire(Calendar cal) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 32. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.